tests/test_filterset.py

```diff
@@ -718,6 +718,27 @@ class Meta:
         self.assertEqual(f.qs.count(), 1)
         self.assertQuerysetEqual(f.qs, [self.alex.pk], lambda o: o.pk)
 
+    def test_fields_set_with_dict(self):
+        class F(FilterSet):
+            class Meta:
+                model = User
+                fields = ['username', 'status', 'is_active', 'first_name']
+                together = [
+                    ({'username': ['status']}),
+                    ({'first_name': ['is_active']}),
+                ]
+
+        f = F({}, queryset=self.qs)
+        self.assertEqual(f.qs.count(), 2)
+        f = F({'username': 'alex'}, queryset=self.qs)
+        self.assertEqual(f.qs.count(), 0)
+        f = F({'username': 'alex', 'status': 1}, queryset=self.qs)
+        self.assertEqual(f.qs.count(), 1)
+        self.assertQuerysetEqual(f.qs, [self.alex.pk], lambda o: o.pk)
+        f = F({'status': 1}, queryset=self.qs)
+        self.assertEqual(f.qs.count(), 1)
+        self.assertQuerysetEqual(f.qs, [self.alex.pk], lambda o: o.pk)
+
```

Comment:
Should the ordering take into account the dependency direction? eg, `together=[{'first_name': ['last_name']}]`. For ordering, should it imply that you must filter by `last_name` before `first_name`? eg, `?o=last_name,first_name` is valid, but `?o=first_name,last_name` is not.